src/parser/xsd/element.js

```diff
@@ -84,6 +84,7 @@ class Element extends XSDElement {
         } else if (child instanceof SimpleType) {
```

Comment:
changes handle the simple-type restrictions only when the simple type definition is placed inside the element directly, like 
```
<xs:element name='e1'>
   <xs:SimpleType>
      <xs:restriction>
```

and not when the simple type is independently declared and referred as a type in the element like,

```
<xs:SimpleType name='type1'>
     <xs:restriction base='xs:decimal'><xs:totaldigits value='4'></xs:restriction> 
</xs:SimplType>
<xs: element name='e1' type='type1' >
```
which is handled in code block https://github.com/strongloop/strong-soap/blob/2c34d198fb93ae1184d09a145be44e7ef354ed09/src/parser/xsd/element.js#L67